src/main/scala/edu/ie3/simona/model/thermal/ThermalGrid.scala

```diff
@@ -594,8 +601,16 @@ object ThermalGrid {
         required: Energy,
         possible: Energy,
     ): ThermalEnergyDemand = {
-      if (possible < required)
-        new ThermalEnergyDemand(possible, possible)
+      if (
+        math.abs(possible.toKilowattHours) < math.abs(
+          required.toKilowattHours
+        ) && math.signum(possible.toKilowattHours) <= math.signum(
+          required.toKilowattHours
+        )
+      )
```

Comment:
If possible would be -2 and required 1 this would not throw an exception, or am I wrong?
Lets go with just positive values (at least for now). So it would be great to check first if req or pos is negative plus if possible is smaller than required. In any case we should throw an an InvalidParameterException with a suitable message.

Reply:
Oh yes, you're right. I removed that if statement now and added a new InvalidParameterException for the cases that possible or required are less than zero.